examples/query/query.go

```diff
@@ -43,33 +43,11 @@ func main() {
 	}
 
 	// define schema
-	fmt.Printf(msgFmt, "create collection `hello_milvus")
-	schema := &entity.Schema{
-		CollectionName: collectionName,
-		Description:    "hello_milvus is the simplest demo to introduce the APIs",
-		AutoID:         false,
-		Fields: []*entity.Field{
-			{
-				Name:       idCol,
-				DataType:   entity.FieldTypeInt64,
-				PrimaryKey: true,
-				AutoID:     false,
-			},
-			{
-				Name:       randomCol,
-				DataType:   entity.FieldTypeDouble,
-				PrimaryKey: false,
-				AutoID:     false,
-			},
-			{
-				Name:     embeddingCol,
-				DataType: entity.FieldTypeFloatVector,
-				TypeParams: map[string]string{
-					entity.TypeParamDim: fmt.Sprintf("%d", dim),
-				},
-			},
-		},
-	}
+	fmt.Printf(msgFmt, fmt.Sprintf("create collection `%s`", collectionName))
+	schema := entity.NewSchema().WithName(collectionName).WithDescription("query_example is a simple demo to demonstrate query API").
+		WithField(entity.NewField().WithName(idCol).WithDataType(entity.FieldTypeInt64).WithIsPrimaryKey(true)).
```

Comment:
As name and type are required, they should be parameters of `NewField` to avoid user missing to fill them

Reply:
The server should check it, but SDK also need to provide better API to avoid users making mistake

Reply:
The SDK doesn't need to check whether the name and type are valid, it only requires the two be provided

Reply:
Since `Field` is still public to user, this API cannot prevent user not filling the name or data type.
As a utility, `WithName` and `WithDataType` demonstrate the detailed field better than `NewField(name, dataType)`